lesson19.js

```diff
@@ -0,0 +1,129 @@
+//Return total days count
+
+//getDaysInYear(year);
+const getDaysInYear = (year) => {
+  return year % 4 === 0 ? 366 : 365;
+};
+console.log(getDaysInYear(2021));
+console.log(getDaysInYear(2020));
+
+//Get day of a year
+//Return day number from date
+
+//getDayNumber(date);
+const getDayNumber = (date) => {
+  return Math.ceil(
+    (Date.parse(date) -
+      Date.parse(new Date(new Date(date).getFullYear(), 0, 1))) /
+      1000 /
+      60 /
+      60 /
+      24
+  );
+};
+
+console.log(getDayNumber("2023-01-12"));
+// 12
+console.log(getDayNumber("2023-02-26"));
+// 57
+console.log(getDayNumber("2023-03-26"));
+
+//Return quarter number from date
+//I quarter: Jan - Mar
+//II quarter: Apr - Jun
+//III quarter: Jul - Sep
+//IV quarter: Oct - Dec
+
+const getQuarters = (date) => {
+  const monthName = new Date(date)
+    .toLocaleString("eng", { month: "long" })
+    .slice(0, 3);
+  const dateNumber = new Date(date).getDate();
+  let quarterName = "";
+  Math.ceil((new Date(date).getMonth() + 1) / 3) === 1
+    ? (quarterName = "I")
+    : Math.ceil((new Date(date).getMonth() + 1) / 3) === 2
+    ? (quarterName = "II")
+    : Math.ceil(new Date(date).getMonth() / 3 + 1) === 3
+    ? (quarterName = "III")
+    : (quarterName = "IV");
+  return `${monthName} ${dateNumber} is ${quarterName} quarter`;
+};
+
+console.log(getQuarters("2023-04-26"));
+// Feb 26 is I quarter
+
+const calcDateDiff = (startDate, endDate) => {
+  const quantityMinutes =
+    (Date.parse(endDate) - Date.parse(startDate)) / 1000 / 60;
+    let result = "";
+  quantityMinutes < 60
+    ? (result = `${quantityMinutes} minutes`)
+    : quantityMinutes >= 60 && quantityMinutes < 1440
+    ? (result = `${(quantityMinutes - (quantityMinutes % 60)) / 60} hours ${
+        quantityMinutes % 60
+      } minutes`)
+    : quantityMinutes >= 1440 && quantityMinutes < 43200
+    ? (result = `${Math.floor(quantityMinutes / 1440)} days`)
+    : quantityMinutes >= 43200 && quantityMinutes < 525600
+    ? (result = `${Math.floor(quantityMinutes / 43200)} month`)
+    : (result = `${Math.floor(quantityMinutes / 525600)} years`);
+  return result;
```

Comment:
good work👍
2  suggestions

1. for big numbers or any  other hardcoded value it's better to create a variable with self-explained name
2. if you don't return value from ternary use if statement